### ISSUE TITLE
App Data path?

### ISSUE THREAD

**Issue by Andy-ABTec** [TV-Rename/tvrename#186]
Will the App Data path be updating to from 2.1 to 2.3 with the new release?

Leaving the old path in place complete with content would make regressing back to the previous version as easy as pie!

I'm guessing it should have changed with the 2.2 release as well but due to all the other stuff going on with thetvdb.com etc got overlooked...

**Comment by mhurwood** [comment]
Probably not a bad idea, but I'm not sure if there's code in there to copy the data across on the first run of a new version.

**Comment by MarkSummerville** [comment]
I'm not sure about this one. We went to great lengths (well some anyway) to make this release backward compatible XML wise (arguably we should have done more). Version 2.1 stems from approx 2008, so we've been on this XML structure for some time. I think we should say that the XML is version 2.1 and the app is 2.3.  In time we'll update the format to use reflection/serialization or some other funky way to store the settings and then we'll bump the storage version.

Basically I don't want us to have to write and test a load of code copying from 2.1 to 2.3 just so that we keep the version numbers in-sync. Further we'll have to deal with the problem every time we move the version number up and deal with migrations from 2.1->2.3, 2.1 ->2.4, etc etc.

**Comment by Andy-ABTec** [comment]
Fair comment - it was only a thought...
Just out of interest what's the reasoning behind ..\Roaming\TVRename\TVRename\2.1? Would it not be less confusing just to stuff everything in ..\Roaming\TVRename\?

**Comment by MarkSummerville** [comment]
I completely agree. It's a legacy thing. Perhaps @sstteevvee can opine?

**Comment by JoeBiellik** [comment]
The reason for ``Roaming\TVRename\TVRename\2.1`` is because currently ``PathManager`` uses ``Application.UserAppDataPath`` which is populated from the current assembly's ``CompanyName\ProductName\ProductVersion`` attributes. Part of the complexity of #187 is that we can't change the ProductVersion without it affecting the paths.

I would personally vote for changing it to ``Roaming\TV Rename``, with the space, to properly reflect the application name. If XML schema versioning is a concern then I would use ``TheTVDB-2.1.xml`` perhaps.

Considering all the recent activity on this project and these potentially breaking changes, have you considered making the next release v3.0.0? It might help existing users know there is new development and give us a chance to make changes with less focus on backward compatibility.

A very small amount of code could be added to check the old directory and copy the files to the new location if the new location has no files at startup.

**Comment by Andy-ABTec** [comment]
No problem it was just a question...
I'm updating the documentation and to me it looked odd - at least I know the reason now 😉

**Comment by MarkSummerville** [comment]
Hi,

Yes I think the next big release (after this one) will be 3 - I'm hoping to put a new UI on it to bring it up to date. I would then split it out to:
```
roaming/TV Rename/logs (log files)
roaming/TV Rename/config (3x XML files
roaming/TV Rename/cache (TVDB.XML)
```
and each would have an archive folder for the old stuff rather than the 0,1,2..9 file sin the same location. Further we could version the XML file properly either with a Schema or just making better use of the existing `<TVRename  Version="2.1">` attribute

**Comment by MarkSummerville** [comment]
Link is here for anyone else who wants to try https://www.dropbox.com/s/6oiweb51rvqlywb/TVRename.exe?dl=0

**Comment by sstteevvee** [comment]
I can confirm that the version number was hard coded so that each upgrade doesn't start with new settings. TVRename was my first .NET app and I learnt the hard way about how it automatically names its settings folder.

I stayed away from using a different folder (and automatically copying old config from the 2.1 one) so that users could downgrade without losing settings, if a new release caused problems.

That said, I think it is worth using the proposed simpler folder name in the future.

**Comment by JoeBiellik** [comment]
I'm currently working on a new installer script for v3 which detects if v2 is installed, copies the config to ``TV Rename\config`` and uninstalls it, before installing the new version.

Doing it this way avoids the need for any extra code.

**Comment by MarkSummerville** [comment]
@JoeBiellik Given the two PRs that are in play I was wondering if it is worth including this in the next release rather than holding off until the v3 release? What are your thoughts? I thought we'd wait as I assumed it'd be a while until we had it all figured out. However you have done a great job refactoring and it looks like we'll be ready now. Might be better to get this out and not have code hanging around?

**Comment by Andy-ABTec** [comment]
Are you talking about the whole shabang or just the configuration files? Either way I'll have to update the docs...

https://andy-abtec.github.io/tvrename/technical

**Comment by JoeBiellik** [comment]
@MarkSummerville I think that really depends on what else we want to include in the last (?) v2 release. If we can get that out soon then I personally think its better to keep these technically breaking changes to v3 but if we think there will be more v2 releases then maybe it makes sense to change this now.

I think a roadmap would be invaluable before we release anything further; I'd like to see exactly what we have changed so far since the last stable build, what else we want in a v2 release and what we plan on changing in v3.

**Comment by MarkSummerville** [comment]
Is this complete with v3? I think it can be closed?

**Comment by JoeBiellik** [comment]
Yes, this is in the v3 branch as #214.
